### ISSUE TITLE
XLNet text generation ability

### ISSUE THREAD

**Issue by Colanim** [huggingface/pytorch-pretrained-BERT#774]
Really appreciate the good work to implement XLNet !

I tried running the [XLNet text generation example](https://github.com/huggingface/pytorch-pretrained-BERT/blob/xlnet/examples/generation_xlnet.py)

But the generated text quality is really low. 

Tricks used by https://github.com/rusiaaman/XLnet-gen needs to be added to the example to generate a good example.

---

... Or is it because the pytorch version of XLNet is not fully working yet ?

**Comment by thomwolf** [comment]
Indeed, I've now added the text padding trick of Aman (add some padding text to have longer inputs) and the quality is really a lot higher.

Will merge the xlnet branch in master and release on Monday.
